packages/sources/bitgo/src/transport/wallet.ts

```diff
@@ -0,0 +1,230 @@
+    signature: string
+  }[]
+  count: number
+  nextBatchPrevId?: string
+}
+
+export class WalletTransport extends SubscriptionTransport<WalletTransportTypes> {
+  name!: string
+  responseCache!: ResponseCache<WalletTransportTypes>
+  requester!: Requester
+  settings!: WalletTransportTypes['Settings']
+
+  async initialize(
+    dependencies: TransportDependencies<WalletTransportTypes>,
+    adapterSettings: WalletTransportTypes['Settings'],
+    endpointName: string,
+    transportName: string,
+  ): Promise<void> {
+    await super.initialize(dependencies, adapterSettings, endpointName, transportName)
+    this.requester = dependencies.requester
+    this.settings = adapterSettings
+  }
+
+  async backgroundHandler(
+    context: EndpointContext<WalletTransportTypes>,
+    entries: RequestParams[],
+  ) {
+    await Promise.all(entries.map(async (param) => this.handleRequest(param)))
+    await sleep(context.adapterSettings.BACKGROUND_EXECUTE_MS)
+  }
+
+  async handleRequest(param: RequestParams) {
+    let response: AdapterResponse<WalletTransportTypes['Response']>
+    try {
+      response = await this._handleRequest(param)
+    } catch (e) {
+      const errorMessage = e instanceof Error ? e.message : 'Unknown error occurred'
+      logger.error(e, errorMessage)
+      response = {
+        statusCode: (e as AdapterError)?.statusCode || 502,
+        errorMessage,
+        timestamps: {
+          providerDataRequestedUnixMs: 0,
+          providerDataReceivedUnixMs: 0,
+          providerIndicatedTimeUnixMs: undefined,
+        },
+      }
+    }
+    await this.responseCache.write(this.name, [{ params: param, response }])
+  }
+
+  async _handleRequest(
+    params: RequestParams,
+  ): Promise<AdapterResponse<WalletTransportTypes['Response']>> {
+    const { chainId, network } = params
+    const coin = params.coin.toLowerCase()
+
+    const providerDataRequestedUnixMs = Date.now()
+
+    const { apiKey, apiEndpoint, apiLimit } = getApiInfo(params.reserve, this.settings)
+
+    // get wallet Ids for a coin
+    const walletIds = await this.fetchWalletIds(coin, apiKey, apiEndpoint, apiLimit)
+
+    // get addresses for each wallet
+    const addresses = await this.fetchWalletAddresses(
+      coin,
+      walletIds,
+      apiKey,
+      apiEndpoint,
+      apiLimit,
+    )
+
+    const result = addresses.map((address) => {
+      return {
+        address,
+        chainId,
+        network,
+      }
+    })
+
+    return {
+      data: {
+        result,
+      },
+      statusCode: 200,
+      result: null,
+      timestamps: {
+        providerDataRequestedUnixMs,
+        providerDataReceivedUnixMs: Date.now(),
+        providerIndicatedTimeUnixMs: undefined,
+      },
+    }
+  }
+
+  async fetchWalletAddresses(
+    coin: string,
+    walletIds: string[],
+    apiKey: string,
+    apiEndpoint: string,
+    apiLimit: number,
+  ) {
+    const addresses: string[] = []
+
+    for (const walletId of walletIds) {
```

Comment:
can we prep all of the requests and fire them at the same time to reduce latency?

Example expectation afaik is ~50 wallets https://chainlink-core.slack.com/archives/C07KYEAS84S/p1725648502950229

Reply:
I am gonna parallel this for loop but not sure how to deal with the hasNext